### PR TITLE
Feature/object conversion to mock table

### PR DESCRIPTION
**Pull request by unby**

### CODE REVIEW

File: DbMocker.Tests/ConversionTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+namespace DbMocker.Tests
+{
+    [TestClass]
+    public class ConversionTests
+    {
+        [TestMethod]
+        public void Mock_EmptyTable_Test()
+        {
+            var mockTable = MockTableExtensions.GetEmptyMockTable<RowDto>();
```
Comment (dvoituron) [inline DbMocker.Tests/ConversionTests.cs:14]: I prefer to create a new typed MockTable using:
```csharp
public class MockTable<T> : MockTable
{
    ...
}
```

So, you will create a new table like that: `var mockTable = new MockTable<RowDto>();`
And you can also use a second possibility using `var mockTable = MockTable.Empty<RowDto>();`

File: DbMocker/ColumnSequenceContainer.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Apps72.Dev.Data.DbMocker
+{
+    /// <summary>
+    /// Container for configuring columns sequence
```
Comment (dvoituron) [inline DbMocker/ColumnSequenceContainer.cs:10]: Can you more explain the goal of this class?

File: DbMocker.Tests/ConversionTests.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            Assert.AreEqual(1, mockTable.Rows.GetLength(0));
+            Assert.AreEqual(3, mockTable.Columns.Length);
+            Assert.IsTrue(mockTable.Columns.Any(a => a.Name.Equals("MaimName")));
+        }
+
+        [TestMethod]
+        public void Mock_NamedTupleConversion_Test2()
+        {
+            var dto = (Id: 1, Name: "Mock", Price: (decimal?)null);
+            var mockTable = dto.ToMockTable((s) => s.Add(k => k.Name, "MaimName"));
```
Comment (dvoituron) [inline DbMocker.Tests/ConversionTests.cs:63]: I found this syntax a little complex. Do you think it's possible to simplify?

File: DbMocker/ColumnSequenceContainer.cs (hunk 2 above)
Comment (unby) [reply DbMocker/ColumnSequenceContainer.cs:10]: 1. We use external types and don't assurance what describe fields and property didn't change order
2. We use some type for different queries with different column set and different column order
3. We want to set several identical names for the columns of one table. It needs for MultiMapping (https://dapper-tutorial.net/result-multi-mapping) when the primary keys of our tables are named the same

File: DbMocker.Tests/ConversionTests.cs (hunk 3 above)
Comment (unby) [reply DbMocker.Tests/ConversionTests.cs:63]: It is impossible. We will lose control.

File: DbMocker.Tests/ConversionTests.cs (hunk 1 above)
Comment (unby) [reply DbMocker.Tests/ConversionTests.cs:14]: Then logics moves to the class' constructor.
I can do it and keep fluent methods.

Comment (unby) [reply DbMocker.Tests/ConversionTests.cs:14]: A lot of logic move to the class' constructor. It isn't ok. 
But  I can make it when we agree on other questions.

File: DbMocker/ColumnSequenceContainer.cs (hunk 2 above)
Comment (unby) [reply DbMocker/ColumnSequenceContainer.cs:10]: @dvoituron, what do You think about configuring columns order?

**Comment by unby** [comment]
I would add function for tables union to one flat table 
(t1 :[c1][c2]) and (t2 :[c1][c3]) => (newTable [c1][c2][c1][c3])
It continue idia of [Multi-Mapping ](https://dapper-tutorial.net/result-multi-mapping)

**Comment by dvoituron** [comment]
The code included in this PR is already integrated in new developments: `MockTable.FromType<T>`

**Pull request closed without merging**